docs/components.ipynb

```diff
@@ -0,0 +1,386 @@
+    "\n",
+    "# Build the instrument model\n",
+    "model = tof.Model(pulse=pulse, detectors=[detector])\n",
+    "model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "75dd1359-d990-482c-b0d6-4a0ae094b0fc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run and plot the rays\n",
+    "model.run()\n",
+    "model.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a2c80ee1-5729-4a09-88c0-3c4e04610772",
+   "metadata": {},
+   "source": [
+    "As expected, the detector sees all the neutrons from the pulse.\n",
+    "Each component in the instrument has a `.plot()` method,\n",
+    "which allows us to quickly visualize histograms of the neutron counts at the detector."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "112124d4-689b-408a-9be6-de1bef8a6205",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "detector.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "34267819-4f83-48c6-b51b-9f13742ac3da",
+   "metadata": {},
+   "source": [
+    "The data itself is available via the `.tofs` and `.wavelengths` properties,\n",
+    "depending on which one you wish to inspect:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bbda2581-ebfa-44ce-ae58-1f4f2db832d9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "detector.tofs.visible.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89ad197a-4304-4cff-a332-e865a2c93f43",
+   "metadata": {},
+   "source": [
+    "## Adding a chopper\n",
+    "\n",
+    "Next, we add a chopper in the beamline,\n",
+    "with a frequency, phase, distance from source,\n",
+    "and a set of open and close angles for the cutouts in the rotating disk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e7e49cac-affd-4c01-860f-03934bc348e9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chopper1 = tof.Chopper(\n",
+    "    frequency=40.0 * Hz,\n",
+    "    open=sc.array(\n",
+    "        dims=['cutout'],\n",
+    "        values=[60.0, 150.0],\n",
+    "        unit='deg',\n",
+    "    ),\n",
+    "    close=sc.array(\n",
+    "        dims=['cutout'],\n",
+    "        values=[90.0, 190.0],\n",
+    "        unit='deg',\n",
+    "    ),\n",
+    "    phase=60.0 * deg,\n",
+    "    distance=8 * meter,\n",
+    "    name=\"Chopper1\",\n",
+    ")\n",
+    "chopper1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6893e0a2-70c9-424f-aeb2-aac010fc02bd",
+   "metadata": {},
+   "source": [
+    "We can directly set this on our existing model, and re-run the simulation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "48909c2a-51e8-4532-a8eb-7c7bbba7544d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.choppers = [chopper1]  # Note that model.choppers needs to be a list\n",
+    "model.run()\n",
```

Comment:
No. You mean like a flag that tells the model it is out of date? It would need to propagate that flag to the components, in case someone inspects the data on the detector after having changed the choppers and not re-run the model.

Reply:
So I changed the mechanism where calling `model.run()` returns a read-only `Result` object that contains all the data and `.plot()` methods.